### PR TITLE
Limit memory

### LINKED ISSUE #6: Limit memory usage to 10MB
The agent should not add more than 10MB of memory at any point.

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #6

### CODE REVIEW

**Comment by bshaffer** [comment]
These changes look awesome :D

**Comment by chingor13** [comment]
@tmatsuo re: stopping capturing variables mid snapshot, we don't actually add any memory to the request - only the refcount is updated. If the snapshot is reported immediately, the refcount decreases and the variable could then be freed, so no extra memory is used. If the snapshot is saved for the end of the request, and we're now over the memory limit, then we prevent further breakpoints from executing for that request.

File: stackdriver_debugger.c [hunk 1]
```diff
@@ -496,3 +522,14 @@ PHP_INI_MH(OnUpdate_stackdriver_debugger_max_time)
...
     return SUCCESS;
 }
+
+/**
+ * Callback for when the user changes the max memory php.ini setting.
+ */
+PHP_INI_MH(OnUpdate_stackdriver_debugger_max_memory)
+{
+    if (new_value != NULL) {
+        STACKDRIVER_DEBUGGER_G(max_memory) = 1000000 * ZEND_STRTOL(ZSTR_VAL(new_value), NULL, 0);
```
Comment (tmatsuo) [inline stackdriver_debugger.c]: Not 1048576?

Comment (chingor13) [reply stackdriver_debugger.c]: You're right, other PHP conversions from use 1024 for 1k, so we'll change to 1048576 (1024*1024)

**Review by tmatsuo — APPROVED** [review]

**Pull request merged**